apps/meteor/app/lib/server/functions/saveCustomFieldsWithoutValidation.ts

```diff
@@ -1,49 +1,66 @@
-import type { IUser, DeepWritable } from '@rocket.chat/core-typings';
+import type { IUser } from '@rocket.chat/core-typings';
+import type { Updater } from '@rocket.chat/models';
 import { Subscriptions, Users } from '@rocket.chat/models';
 import { Meteor } from 'meteor/meteor';
-import type { UpdateFilter } from 'mongodb';
 
 import { trim } from '../../../../lib/utils/stringUtils';
 import { settings } from '../../../settings/server';
 import { notifyOnSubscriptionChangedByUserIdAndRoomType } from '../lib/notifyListener';
 
-export const saveCustomFieldsWithoutValidation = async function (userId: string, formData: Record<string, any>): Promise<void> {
-	if (trim(settings.get('Accounts_CustomFields')) !== '') {
-		let customFieldsMeta;
-		try {
-			customFieldsMeta = JSON.parse(settings.get('Accounts_CustomFields'));
-		} catch (e) {
-			throw new Meteor.Error('error-invalid-customfield-json', 'Invalid JSON for Custom Fields');
-		}
+const getCustomFieldsMeta = function (customFieldsMeta: string) {
+	try {
+		return JSON.parse(customFieldsMeta);
+	} catch (e) {
+		throw new Meteor.Error('error-invalid-customfield-json', 'Invalid JSON for Custom Fields');
+	}
+};
+export const saveCustomFieldsWithoutValidation = async function (
+	userId: string,
+	formData: Record<string, any>,
+	_updater?: Updater<IUser>,
+): Promise<void> {
+	const customFieldsSetting = settings.get<string>('Accounts_CustomFields');
+	if (!customFieldsSetting || trim(customFieldsSetting).length === 0) {
+		return;
+	}
+
+	// configured custom fields in setting
+	const customFieldsMeta = getCustomFieldsMeta(customFieldsSetting);
+
+	const customFields: Record<string, any> = Object.keys(customFieldsMeta).reduce(
+		(acc, currentValue) => ({
+			...acc,
+			[currentValue]: formData[currentValue],
+		}),
+		{},
+	);
 
-		const customFields: Record<string, any> = {};
-		Object.keys(customFieldsMeta).forEach((key) => {
-			customFields[key] = formData[key];
-		});
-		await Users.setCustomFields(userId, customFields);
+	const updater = _updater || Users.getUpdater();
 
-		// Update customFields of all Direct Messages' Rooms for userId
-		const setCustomFieldsResponse = await Subscriptions.setCustomFieldsDirectMessagesByUserId(userId, customFields);
-		if (setCustomFieldsResponse.modifiedCount) {
-			void notifyOnSubscriptionChangedByUserIdAndRoomType(userId, 'd');
+	updater.set('customFields', customFields);
```

Comment:
```suggestion
```
You're setting customFields as a whole here, then still iterating the object to set each field individually.